sogs/model/room.py

```diff
@@ -1363,7 +1399,27 @@ def unpin_all(self, admin: User):
             app.logger.warning("Unable to unpin all messages from {self}: {admin} is not an admin")
             raise BadPermission()
 
-        count = query("DELETE FROM pinned_messages WHERE room = :r", r=self.id).rowcount
+        with db.transaction():
+            if config.UPLOAD_DEFAULT_EXPIRY_DAYS != 15:
```

Comment:
why are we checking this in this manner?

Reply:
It was an optimization, but a premature one after thinking about it, so removed.